lib/sast.ml

```diff
@@ -6,7 +6,7 @@ and sx =
   | SBoolLit of bool
   | SCharLit of char
   | SStringLit of string
-  | SId of string
+  | SId of string * int
```

Comment:
can u explain the significance of the int component?

Reply:
oh nvm is it the index?

Reply:
yessir, it's like the index of the scope where the Id lives. we could write IR without it but this allows us to avoid more scanning over the `scopes` list.

Reply:
awesome